listenbrainz_spark/recommendations/train_models.py

```diff
@@ -69,8 +80,88 @@ def preprocess_data(playcounts_df):
     return training_data, validation_data, test_data
 
 
-def train(training_data, validation_data, num_validation, ranks, lambdas, iterations, alpha):
-    """ Train the data and get models as per given parameters i.e. ranks, lambdas and iterations.
+def generate_model_id():
+    """ Generate model id.
+    """
+    return '{}-{}'.format(config.MODEL_ID_PREFIX, uuid.uuid4())
+
+
+def get_best_model_path(model_id):
+    """ Get path to save best model
+
+        Args:
+            model_id (str): Model identification string of best model.
+
+        Returns:
+            path to save best model.
+    """
+
+    return config.HDFS_CLUSTER_URI + path.DATA_DIR + '/' + model_id
+
+
+def get_latest_dataframe_id(dataframe_metadata_df, best_model_metadata):
```

Comment:
This function is named get_ ... but it doesn't return anything. Nor does the docstring say that data in best_model_metadata will be updated.

Reply:
oops! should be `update_best_model_metadata`?

Reply:
Sounds better, yes.